assert/assertions_test.go

```diff
@@ -482,146 +465,101 @@ func TestExactly(t *testing.T) {
-		t.Error("NotEqualValues should return true")
-	}
-	if !NotEqualValues(mockT, 123, 1234) {
-		t.Error("NotEqualValues should return true")
-	}
-	if !NotEqualValues(mockT, 123.5, 123.55) {
-		t.Error("NotEqualValues should return true")
-	}
-	if !NotEqualValues(mockT, []byte("Hello World"), []byte("Hello World!")) {
-		t.Error("NotEqualValues should return true")
-	}
-	if !NotEqualValues(mockT, nil, new(AssertionTesterConformingObject)) {
-		t.Error("NotEqualValues should return true")
-	}
-	if NotEqualValues(mockT, nil, nil) {
-		t.Error("NotEqualValues should return false")
-	}
-	if NotEqualValues(mockT, "Hello World", "Hello World") {
-		t.Error("NotEqualValues should return false")
-	}
-	if NotEqualValues(mockT, 123, 123) {
-		t.Error("NotEqualValues should return false")
-	}
-	if NotEqualValues(mockT, 123.5, 123.5) {
-		t.Error("NotEqualValues should return false")
-	}
-	if NotEqualValues(mockT, []byte("Hello World"), []byte("Hello World")) {
-		t.Error("NotEqualValues should return false")
-	}
-	if NotEqualValues(mockT, new(AssertionTesterConformingObject), new(AssertionTesterConformingObject)) {
-		t.Error("NotEqualValues should return false")
-	}
-	if NotEqualValues(mockT, &struct{}{}, &struct{}{}) {
-		t.Error("NotEqualValues should return false")
-	}
+	cases := []TestCase{
+		// cases that are expected not to match
+		{"Hello World", "Hello World!", true},
+		{123, 1234, true},
+		{123.5, 123.55, true},
+		{[]byte("Hello World"), []byte("Hello World!"), true},
+		{nil, new(AssertionTesterConformingObject), true},
 
-	// Special cases where NotEqualValues behaves differently
-	funcA := func() int { return 23 }
-	funcB := func() int { return 42 }
-	if !NotEqualValues(mockT, funcA, funcB) {
-		t.Error("NotEqualValues should return true")
-	}
-	if !NotEqualValues(mockT, int(10), int(11)) {
-		t.Error("NotEqualValues should return true")
-	}
-	if NotEqualValues(mockT, int(10), uint(10)) {
-		t.Error("NotEqualValues should return false")
-	}
-	if NotEqualValues(mockT, struct{}{}, struct{}{}) {
-		t.Error("NotEqualValues should return false")
+		// cases that are expected to match
+		{nil, nil, false},
+		{"Hello World", "Hello World", false},
+		{123, 123, false},
+		{123.5, 123.5, false},
+		{[]byte("Hello World"), []byte("Hello World"), false},
+		{new(AssertionTesterConformingObject), new(AssertionTesterConformingObject), false},
+		{&struct{}{}, &struct{}{}, false},
+
+		// Different behaviour from NotEqual()
+		{func() int { return 23 }, func() int { return 24 }, true},
+		{int(10), int(11), true},
+		{int(10), uint(10), false},
+
+		{struct{}{}, struct{}{}, false},
 	}
-}
 
-type A struct {
-	Name, Value string
+	for _, c := range cases {
+		res := NotEqualValues(mockT, c.expected, c.actual)
+
+		if res != c.result {
+			t.Errorf("NotEqualValues(%#v, %#v) should return %#v", c.expected, c.actual, c.result)
+		}
+	}
 }
 
-func TestContains(t *testing.T) {
+func TestContainsNotContains(t *testing.T) {
```

Comment:
In this case, in my opinion, it makes sense to combine TestContains and TestNoContains with the same set of cases to avoid duplocation. Also, if a case ever gets added we get it for free for the opposite test.